### PR TITLE
Github Actions Integration tests: Add latest as fallback

### PR DESCRIPTION
**Pull request by jaypontes**

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -19,14 +19,25 @@ jobs:
...
       if: inputs.trigger == 'pull-request'
-      run: | # If the image with PR branch's version as tag doesn't exist, use the one in main that should already exist
+      run: | # If the image with PR branch's version as tag doesn't exist, use the one in main that should already exist (fallback on 0.4.56 incase neither exists)
         git fetch -q origin main:main
-        docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-cd-service:$(make version) > /dev/null && \
-          echo "IMAGE_TAG_CD=$(make version)" >> $GITHUB_ENV || \
+
+        if docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-cd-service:$(make version) > /dev/null; then
+          echo "IMAGE_TAG_CD=$(make version)" >> $GITHUB_ENV
+        elif docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-cd-service:$(git describe --always --long --tags main) > /dev/null; then
```
Comment (tameremad) [inline .github/workflows/integration-tests.yml:27]: Maybe we should remove the `--long` option to get the image tag? 

```
$ git describe --always --long --tags main
0.4.56-0-gdc175cb   // which doesn't exist
```
```
$ git describe --always --tags main
0.4.56   // which is okay
```
Or do I miss something?

Comment (jaypontes) [reply .github/workflows/integration-tests.yml:27]: 😮 
Awesome!! Wasn't aware of this. love it! :)

**Review by tameremad — APPROVED** [review]
:+1: great!

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -19,14 +19,25 @@ jobs:
         fetch-depth: 0
     - name: Setup PR environment
       if: inputs.trigger == 'pull-request'
-      run: | # If the image with PR branch's version as tag doesn't exist, use the one in main that should already exist
+      run: | # If the image with PR branch's version as tag doesn't exist, use the one in main that should already exist (fallback on 0.4.56 incase neither exists)
```
Comment (tameremad) [inline .github/workflows/integration-tests.yml:22]: Maybe also edit the readme first.

Comment (jaypontes) [reply .github/workflows/integration-tests.yml]: Very true, done in the next commit!

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -19,14 +19,27 @@ jobs:
...
+          echo "IMAGE_TAG_CD=$(git describe --always --tags main)" >> $GITHUB_ENV
+        fi
+
+        if docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-frontend-service:$(make version) > /dev/null; then
+          echo "IMAGE_TAG_FRONTEND=$(make version)" >> $GITHUB_ENV
+        elif docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-frontend-service:$(git describe --always --long --tags main); then
           echo "IMAGE_TAG_FRONTEND=$(git describe --always --long --tags main)" >> $GITHUB_ENV
+        elif docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-frontend-service:$(git describe --always --tags main); then
+          echo "IMAGE_TAG_FRONTEND=$(git describe --always --tags main)" >> $GITHUB_ENV
+        fi
```
Comment (sven-urbanski-freiheit-com) [inline .github/workflows/integration-tests.yml:41]: what if none of the if-conditions match?
Please add an "else". I imagine in the "else" we want to exit with an error.

File: .github/workflows/integration-tests.yml [hunk 4]
```diff
@@ -19,14 +19,27 @@ jobs:
...
+
+        if docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-cd-service:$(make version) > /dev/null; then
+          echo "IMAGE_TAG_CD=$(make version)" >> $GITHUB_ENV
+        elif docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-cd-service:$(git describe --always --long --tags main) > /dev/null; then
           echo "IMAGE_TAG_CD=$(git describe --always --long --tags main)" >> $GITHUB_ENV
-        docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-frontend-service:$(make version) > /dev/null && \
-          echo "IMAGE_TAG_FRONTEND=$(make version)" >> $GITHUB_ENV || \
+        elif docker manifest inspect ${IMAGE_REGISTRY}/kuberpult-cd-service:$(git describe --always --tags main) > /dev/null; then
+          echo "IMAGE_TAG_CD=$(git describe --always --tags main)" >> $GITHUB_ENV
+        fi
```
Comment (sven-urbanski-freiheit-com) [inline .github/workflows/integration-tests.yml:33]: same here about "else"

**Pull request merged**